src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -1479,6 +1470,150 @@ static void Main()
+    }
+}
+" + trivial2uple;
+
+            var source = @"
+class C3
+{
+    public static void Main()
+    {
+        System.Console.Write(C1.M().Item1 + "" "");
+        System.Console.Write(C1.M().a + "" "");
+        System.Console.Write(C1.M().Item2 + "" "");
+        System.Console.Write(C1.M().b + "" "");
+        System.Console.Write(C2.M().Item1 + "" "");
+        System.Console.Write(C2.M().c + "" "");
+        System.Console.Write(C2.M().Item2 + "" "");
+        System.Console.Write(C2.M().d);
+    }
+}
+";
+            var comp1 = CreateCompilationWithMscorlib(source1);
+            var comp2 = CreateCompilationWithMscorlib(source2);
+            var comp = CompileAndVerify(source, expectedOutput: @"1 1 2 2 3 3 4 4", additionalRefs: new[] { new CSharpCompilationReference(comp1), new CSharpCompilationReference(comp2) });
+        }
+
+        [Fact]
+        public void AmbiguousTupleTypesInCompilation()
+        {
+            var source = @"
+class C3
+{
+    public static void Main()
+    {
+        (int, int) x = (1, 1);
+    }
+}
+";
+            var comp1 = CreateCompilationWithMscorlib(trivial2uple);
+            var comp2 = CreateCompilationWithMscorlib(trivial2uple);
+
+            var comp = CompileAndVerify(source, additionalRefs: new[] { new CSharpCompilationReference(comp1), new CSharpCompilationReference(comp2) });
+            comp.VerifyDiagnostics(
+                // warning CS1685: The predefined type 'ValueTuple<T1, T2>' is defined in multiple assemblies in the global alias; using definition from '__assembly_guid__, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
+                Diagnostic(ErrorCode.WRN_MultiplePredefTypes).WithArguments("System.ValueTuple<T1, T2>", comp1.AssemblyName + ", Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(1, 1)
+                                );
+        }
+
+        [Fact]
+        public void LocalTupleTypeWinsWhenTupleTypesInCompilation()
+        {
+            var source1 = @"
+public class C1
+{
+    public static (int a, int b) M()
+    {
+        return (1, 2);
+    }
+}
+" + trivial2uple;
+
+            var source2 = @"
+public class C2
+{
+    public static (int c, int d) M()
+    {
+        return (3, 4);
+    }
+}
+" + trivial2uple;
+
+            var source = @"
+class C3
+{
+    public static void Main()
+    {
+        System.Console.Write(C1.M().Item1 + "" "");
+        System.Console.Write(C1.M().a + "" "");
+        System.Console.Write(C1.M().Item2 + "" "");
+        System.Console.Write(C1.M().b + "" "");
+        System.Console.Write(C2.M().Item1 + "" "");
+        System.Console.Write(C2.M().c + "" "");
+        System.Console.Write(C2.M().Item2 + "" "");
+        System.Console.Write(C2.M().d + "" "");
+
+        var x = (e: 5, f: 6);
+        System.Console.Write(x.Item1 + "" "");
+        System.Console.Write(x.e + "" "");
+        System.Console.Write(x.Item2 + "" "");
+        System.Console.Write(x.f);
+
+    }
```

Comment:
We don't verify that type of x is the one defined in current assembly. This can be done by comparing x.GetType().Assembly with typeof(C3).Assembly and printing the result.